browser/brave_rewards/checkout_dialog.cc

```diff
@@ -0,0 +1,317 @@
+  for (size_t i = 0; i < display_items.size(); i++) {
+    DCHECK((*display_items[i])->sku.has_value());
+    auto item = ledger::type::SKUOrderItem::New();
+
+    item->sku = (*display_items[i])->sku.value();
+    item->quantity = 1;
+    items_.push_back(std::move(item));
+  }
+
+  auto callback = base::BindOnce(&OnSKUProcessed, request_);
+
+  rewards_service->ProcessSKU(
+      std::move(items_), ledger::constant::kWalletUphold, std::move(callback));
+}
+
+void CheckoutDialogHandler::GetRewardsParameters(const base::ListValue* args) {
+  if (auto* service = GetRewardsService()) {
+    AllowJavascript();
+    service->GetRewardsParameters(
+        base::Bind(&CheckoutDialogHandler::GetRewardsParametersCallback,
+                   weak_factory_.GetWeakPtr()));
+  }
+}
+
+void CheckoutDialogHandler::GetRewardsParametersCallback(
+    ledger::type::RewardsParametersPtr parameters) {
+  if (!IsJavascriptAllowed()) {
+    return;
+  }
+
+  base::Value data(base::Value::Type::DICTIONARY);
+  if (parameters) {
+    data.SetDoubleKey("rate", parameters->rate);
+    data.SetDoubleKey("lastUpdated", base::Time::Now().ToJsTimeIgnoringNull());
+  }
+  FireWebUIListener("rewardsParametersUpdated", data);
+}
+
+void CheckoutDialogHandler::OnGetWalletBalance(const base::ListValue* args) {
+  if (auto* service = GetRewardsService()) {
+    AllowJavascript();
+    service->FetchBalance(
+        base::BindOnce(&CheckoutDialogHandler::FetchBalanceCallback,
+                       weak_factory_.GetWeakPtr()));
+  }
+}
+
+void CheckoutDialogHandler::FetchBalanceCallback(
+    const ledger::type::Result result,
+    ledger::type::BalancePtr balance) {
+  if (!IsJavascriptAllowed()) {
+    return;
+  }
+
+  base::Value data(base::Value::Type::DICTIONARY);
+  data.SetDoubleKey("total", balance->total);
+
+  FireWebUIListener("walletBalanceUpdated", data);
+}
+
+void CheckoutDialogHandler::GetExternalWallet(const base::ListValue* args) {
+  if (auto* service = GetRewardsService()) {
+    AllowJavascript();
+    service->GetExternalWallet(
+        base::BindOnce(&CheckoutDialogHandler::GetExternalWalletCallback,
+                       weak_factory_.GetWeakPtr()));
+  }
+}
+
+void CheckoutDialogHandler::GetExternalWalletCallback(
+    const ledger::type::Result result,
+    ledger::type::ExternalWalletPtr wallet) {
+  if (!IsJavascriptAllowed()) {
+    return;
+  }
+
+  base::Value data(base::Value::Type::DICTIONARY);
+
+  if (wallet) {
+    data.SetIntKey("status", static_cast<int>(wallet->status));
+  }
+  FireWebUIListener("externalWalletUpdated", data);
+}
+
+void ShowCheckoutDialog(base::WeakPtr<PaymentRequest> request) {
+  WebContents* initiator = request->web_contents();
+  Profile* profile =
+      Profile::FromBrowserContext(initiator->GetBrowserContext());
+
+  // BAT payment method only works for verified publishers
+  auto* service = brave_rewards::RewardsServiceFactory::GetForProfile(profile);
```

Comment:
You're not checking RewardsService for null and if it is null you can skip the whole thing, right?

Reply:
resolved